### ISSUE TITLE
New Rule: Single Line Per Property

### ISSUE THREAD

**Issue by BPScott** [sasstools/sass-lint#283]
Add rule to enforce each property (/ `@include` / `@extends`) within a selector exists on a single line. As per the [existing scss-lint SingleLinePerProperty rule](https://github.com/brigade/scss-lint/blob/master/lib/scss_lint/linter/README.md#singlelineperproperty).

Allowed:

```scss
p {
  margin: 0;
  padding: 0;
}
```

Disallowed:

```scss
p {
  margin: 0; padding: 0;
}
```

There should be a `allow_single_line_rule_sets` option that defaults to `true` to allow the usage of single-line rule sets, to make the following allowed:

```scss
p { margin: 0; padding: 0; }
```

As an extension to the exisiting functionality I think it could be useful to add an additional configuration option `maxPropertiesForSingleLineRuleSet` to allow users to limit the amount of properties within a single-line ruleset, defaulting to `Infinity`. For instance if `maxPropertiesForSingleLineRuleSet` was set to 1 then `p { margin: 0; padding: 0; }` would be invalid, but `p { margin: 0; }` would be acceptable. This would allow people to limit these single-line rulesets to only be used in cases where the number of properties won't affect readability.

**Comment by bgriffith** [comment]
Surprised we don't have this one yet!

**Comment by BPScott** [comment]
Ah, dangit, after looking through the existing rules docs, we've already got [`one-declaration-per-line`](https://github.com/sasstools/sass-lint/blob/develop/docs/rules/one-declaration-per-line.md) which does this. Sorry.

Though it looks like it's missing the `allow_single_line_rule_sets` option, so that'd still be useful to add IMO. Shall I rewrite the title/description or close as invalid and fire up a new ticket? (related: looking back on this it seems my option naming convention is all over the bloody shop, sometimes snake_case, sometimes camelCase, when sass-lint's actual convention is dashed-case)

Is it worth renaming rule to match scss-lint? Or are you not that worried about compatibility.

**Comment by DanPurdy** [comment]
I think this actually raises a valid point.. I'm wondering why we went with `one-declaration-per-line` and then `single-line-per-selector` naming convention, should probably rename the former. It confused me too!

let's close this ticket and fire up a new one if you don't mind.

We should hopefully be getting `1.3.0` out this weekend and there are a LOT of fixes and changes included.

**Comment by BPScott** [comment]
Closing this ticket in favour of:

* #288 for discussion of renaming `one-declaration-per-line` to `single-line-per-property`
* #287 for adding the `allow-single-line-rule-sets` option to `one-declaration-per-line`
